### PR TITLE
Added matvec/rmatvec count

### PR DESCRIPTION
**Pull request by mrava87**
This PR introduces a new decorator `count` which is used to count the number of matvec/rmatvec/matmat/rmatmat performed by a single operator. These counts are stored in the following variables `matvec_count`,  `rmatvec_count`, `matmat_count`, `rmatmat_count` and can be restarted using the method `reset_count`

### CODE REVIEW

File: pylops/utils/decorators.py [hunk 1]
```diff
@@ -121,3 +121,42 @@ def wrapper(self, x):
     if func is not None:
         return decorator(func)
     return decorator
+
+
+def count(f=None, forward=True, mat=False):
```
Comment (cako) [inline pylops/utils/decorators.py:126]: Was it a design choice to not derive forward/mat based on the `f.__name__` as was done in the `reshaped` decorator? If it wasn't and you think it's a good idea, let me know and I can implement it.

File: pylops/utils/decorators.py [hunk 2]
```diff
@@ -121,3 +121,42 @@ def wrapper(self, x):
...
+                    self.matvec_count -= x.shape[-1]
+                else:
+                    self.matvec_count += 1
+            else:
+                if mat:
+                    self.rmatmat_count += 1
+                    self.rmatvec_count -= x.shape[-1]
+                else:
+                    self.rmatvec_count += 1
+            y = f(self, x)
```
Comment (cako) [inline pylops/utils/decorators.py:155]: I think it's better to put this line in the beginning of the wrapper. If for some reason `f` errors, the linear operator will have the wrong count.

**Review by cako — CHANGES_REQUESTED** [review]

File: pylops/utils/decorators.py (hunk 2 above)
Comment (mrava87) [reply pylops/utils/decorators.py:155]: Agree :)

File: pylops/utils/decorators.py (hunk 1 above)
Comment (mrava87) [reply pylops/utils/decorators.py:126]: Mmh not really. I think it makes sense to try to be as consistent as possible between different routines in the library, so I fixed it. Have a look and let me know if you agree.

Question: shall I still be verbose in `LinearOperator` or not (I mean shall I just use @count or specify the parameters? What do you suggest? For now I don't change it, but I tried and the new code would work as well by doing the internal evaluation based on `f.__name__`

Comment (cako) [reply pylops/utils/decorators.py]: Awesome, the change looks good. Regarding your question, personally I would remove all the arguments, but that's my personal preference.

**Review by cako — APPROVED** [review]
Fee free to merge

**Pull request merged**